simple/stats/events_importer.py

```diff
@@ -0,0 +1,280 @@
+
+
+class EventsImporter(Importer):
+  """Imports a single events input file.
+    """
+
+  def __init__(self, input_fh: FileHandler, db: Db,
+               debug_resolve_fh: FileHandler, reporter: FileImportReporter,
+               nodes: Nodes) -> None:
+    self.input_fh = input_fh
+    self.db = db
+    self.debug_resolve_fh = debug_resolve_fh
+    self.reporter = reporter
+    self.nodes = nodes
+    self.input_file_name = self.input_fh.basename()
+    self.config = nodes.config
+    self.entity_type = self.config.entity_type(self.input_file_name)
+    self.ignore_columns = self.config.ignore_columns(self.input_file_name)
+    self.provenance = self.nodes.provenance(self.input_file_name).id
+
+    self.event_type = self.config.event_type(self.input_file_name)
+    assert self.event_type, f"Event type must be specified: {self.input_file_name}"
+
+    self.id_column = self.config.id_column(self.input_file_name)
+
+    self.df = pd.DataFrame()
+    self.debug_resolve_df = None
+
+  def do_import(self) -> None:
+    self.reporter.report_started()
+    try:
+      self._read_csv()
+      self._drop_ignored_columns()
+      self._sanitize_values()
+      self._resolve_entities()
+      self._parse_dates()
+      self._rename_columns()
+      self._write_event_triples()
+      self._write_observations()
+      self.reporter.report_success()
+    except Exception as e:
+      self.reporter.report_failure(str(e))
+      raise e
+
+    self._write_debug_csvs()
+
+  def _read_csv(self) -> None:
+    # Read CSVs with the following behaviors:
+    # - Set 1st column (i.e. the entity column) to type str (so that geoIds like "01" are not treated as ints and converted to 1)
+    # - Strip leading whitespaces
+    # - Treat comma as a thousands separator
+    self.df = pd.read_csv(self.input_fh.read_string_io(),
+                          dtype={0: str},
+                          skipinitialspace=True,
+                          thousands=",")
+    logging.info("Read %s rows.", self.df.index.size)
+
+  def _drop_ignored_columns(self):
+    if self.ignore_columns:
+      self.df.drop(columns=self.ignore_columns, axis=1, inplace=True)
+
+  def _sanitize_values(self):
+    # Convert to best possible dtypes (i.e. keep ints as ints even when some values are NaN)
+    self.df = self.df.convert_dtypes()
+    # Set date field to type str.
+    self.df = self.df.astype({self.df.columns[1]: str})
+
+  def _rename_columns(self) -> None:
+    renamed = {}
+    # Rename dcid and date columns
+    renamed[self.df.columns[0]] = constants.COLUMN_DCID
+    renamed[self.df.columns[1]] = constants.COLUMN_DATE
+
+    # Rename property columns to their IDs
+    property_column_names = self.df.columns[2:]
+    property_ids = [
+        self.nodes.property(property_column_name).dcid
+        for property_column_name in property_column_names
+    ]
+    renamed.update({
+        col: id for col, id in zip(property_column_names, property_ids)
+    })
```

Comment:
merge these lines to:
for col in property_column_names:
  renamed.update({col: self.nodes.property(col).dcid})

Reply:
Let me do this and other panda updates in a separate PR and ensure all tests pass as-is.